pg-extend/src/pg_datum.rs

```diff
@@ -211,6 +212,111 @@ impl<'s> TryFromPgDatum<'s> for String {
     }
 }
 
+// FIXME: this impl is copy-pasta'd from that of Text. Could dedup with macro.
+impl<'s> From<ByteA<'s>> for PgDatum<'s> {
+    fn from(value: ByteA<'s>) -> Self {
+        let ptr = unsafe { value.into_ptr() };
+        PgDatum(Some(ptr as Datum), PhantomData)
+    }
+}
+
+// FIXME: this impl is copy-pasta'd from that of Text. Could dedup with macro.
+impl<'s> TryFromPgDatum<'s> for ByteA<'s> {
+    fn try_from<'mc>(
+        memory_context: &'mc PgAllocator,
+        datum: PgDatum<'mc>,
+    ) -> Result<Self, &'static str>
+    where
+        Self: 's,
+        'mc: 's,
+    {
+        if let Some(datum) = datum.0 {
+            let text_ptr = datum as *const pg_sys::bytea;
+
+            unsafe { Ok(ByteA::from_raw(memory_context, text_ptr as *mut _)) }
+        } else {
+            Err("datum was NULL")
+        }
+    }
+}
+
+impl<'s> TryFromPgDatum<'s> for &[u8] {
```

Comment:
I think we want `&'s [u8]` here, right? and then we should make sure that `'s` is bound by `'mc`, right? in other words, this byte array should be bounded by the lifetime of the memory context that encapsulates it.